avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGeneratorBase.java

```diff
@@ -60,22 +61,25 @@ protected static int nextUniqueInt() {
   public abstract FastSerializer<T> generateSerializer();
 
   @SuppressWarnings("unchecked")
-  protected Class<FastSerializer<T>> compileClass(final String className) throws IOException, ClassNotFoundException {
+  protected Class<FastSerializer<T>> compileClass(final String className, Set<String> knownUsedFullyQualifiedClassNameSet)
```

Comment:
Looks like this class and `FastDeserializerGeneratorBase` have a lot in common. We should consider refactoring the code so they share a common base class or share utility functions...

I don't think it's critical to do this as part of this commit, but we should at least leave a TODO comment.

I leave it up to you whether you want to refactor right away or just leave the TODO (unless you disagree and think the two classes ought to stay separate?)

Reply:
You are right, and this situation has been there from the beginning, and we haven't optimized the code structure yet.
I could leave a TODO.